src/util.ts

```diff
@@ -38,12 +39,77 @@ export function nop() {
 /**
  * Find the tsconfig.json, read it, and return parsed contents.
  * @param rootDir Directory where the tsconfig.json should be found.
+ * If the tsconfig.json file has an "extends" field hop down the dependency tree until it ends
  */
+
+
+export interface ConfigFile {
+  files?: string[];
+  compilerOptions?: {};
+  include?: string[];
+  exclude?: string[];
+}
+
 export async function getTSConfig(
-    rootDir: string, customReadFilep?: ReadFileP): Promise<{}> {
+    rootDir: string, customReadFilep?: ReadFileP): Promise<ConfigFile> {
+  // array to hold accessed files, used to check for circular references
+
+
+
   const tsconfigPath = path.join(rootDir, 'tsconfig.json');
+
   customReadFilep = customReadFilep || readFilep;
   const json = await customReadFilep(tsconfigPath, 'utf8');
-  const contents = JSON.parse(json);
-  return contents;
+  let contents = JSON.parse(json);
+
+  const readArr = ['tsconfig.json'];
+  if (contents['extends']) {
+    const extension =
+        await getExtension(contents['extends'], customReadFilep, readArr);
+    contents = combineTSConfig(extension, contents);
+  }
+  return Promise.resolve(contents);
 }
+
+async function getExtension(
```

Comment:
I would consider calling this `getBase` instead, because the the config file at `filePath` overrides the config that it extends, not the other way around.

Reply:
👍